addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java

```diff
@@ -0,0 +1,354 @@
+                                                || lcFileName.endsWith(".yml"));
+                                    }
+
+                                    @Override
+                                    public String getDescription() {
+                                        return Constant.messages.getString(
+                                                "automation.panel.load.yaml");
+                                    }
+                                });
+                        int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+                        if (rc == JFileChooser.APPROVE_OPTION) {
+                            ext.getParam()
+                                    .setPlanDirectory(
+                                            chooser.getSelectedFile()
+                                                    .getParentFile()
+                                                    .getAbsolutePath());
+                            loadPlan(chooser.getSelectedFile());
+                        }
+                    });
+        }
+        return loadPlanButton;
+    }
+
+    private void loadPlan(File f) {
+        try {
+            setCurrentPlan(ext.loadPlan(f));
+            ext.registerPlan(currentPlan);
+            AutomationProgress progress = currentPlan.getProgress();
+            this.getOutputArea().setText(listToStr(progress.getAllMessages()));
+            if (progress.hasErrors()) {
+                View.getSingleton()
+                        .showWarningDialog(
+                                Constant.messages.getString(
+                                        "automation.panel.load.error",
+                                        listToStr(progress.getErrors())));
+            } else if (progress.hasWarnings()) {
+                View.getSingleton()
+                        .showWarningDialog(
+                                Constant.messages.getString(
+                                        "automation.panel.load.warning",
+                                        listToStr(progress.getWarnings())));
+            }
+
+        } catch (Exception e) {
+            View.getSingleton()
+                    .showWarningDialog(
+                            Constant.messages.getString(
+                                    "automation.panel.load.failed", e.getMessage()));
+        }
+    }
+
+    public void setCurrentPlan(AutomationPlan plan) {
+        currentPlan = plan;
+        getTreeModel().setPlan(currentPlan);
+        getRunPlanButton().setEnabled(currentPlan != null);
+    }
+
+    private JTabbedPane getTabbedPane() {
+        if (this.tabbedPane == null) {
+            this.tabbedPane = new JTabbedPane();
+            this.tabbedPane.addTab("Plan", this.getPlanScrollpane());
+            this.tabbedPane.addTab("Output", getOutputScrollpane());
+        }
+        return this.tabbedPane;
+    }
+
+    private JScrollPane getPlanScrollpane() {
+        if (planScrollpane == null) {
```

Comment:
I think we can avoid these null checks by initializing in the constructor.

Reply:
As above.